src/librustdoc/formats/cache.rs

```diff
@@ -77,6 +79,9 @@ pub(crate) struct Cache {
     /// Cache of where documentation for primitives can be found.
     pub(crate) primitive_locations: FxHashMap<clean::PrimitiveType, DefId>,
 
+    /// Used for [`Cache::intern_item_id`] to create [`rustdoc_json_types::Id`]s
+    item_id_interner: Lock<FxIndexSet<ItemId>>,
```

Comment:
Why are we using an `IndexSet`, instead of a `HashMap<ItemId, rustdoc_json_types::Id>` here? The latter seems far clearer (in my mind, at least).

Reply:
IndexSet returns the index of the item that's been added, with a HashMap it'd need an additional counter attached to it that'd be incremented only if the Id hasn't been added to the map yet

Reply:
You can just use the `.len()` for this (which is what [`IndexMap`](https://github.com/indexmap-rs/indexmap/blob/abdd155f73331040991f24eaff6f37976097711e/src/map/core/raw.rs#L70) does internally). Yes this means we need to implement a bit more logic in rustdoc, but it make the nature as a Map clearer (rather than being a set), and means you don't have to pay the overhead of `IndexMap` (vs hashbrown's HashMap)